tests/MongoDB.Driver.Legacy.Tests/Builders/UpdateBuilderTests.cs

```diff
@@ -838,26 +838,25 @@ public void TestRename()
             Assert.Equal(expected, update.ToJson());
         }
 
-// tests fail on .NET Core 2.1 with out of order key-value-pairs        
-#if !NETCOREAPP2_1        
```

Comment:
I've rewritten this test to not be depended on this issue which can be simply fixed

Reply:
Order is important in BsonDocuments: should we simply be changing the test so that we don't care about order? @rstam

Reply:
@DmitryLukyanov can you investigate this further?

It seems like Equal should have worked (perhaps a few formatting issues need to be worked out)?

Reply:
If the fields really are in a different order on .NET Core it must be something in the order of the fields in the BsonClassMap.

In which case `BeEquivalentTo` is the right choice.

Reply:
IIRC this test passes in .NET Core 1.0 but fails in .NET Core 2.1. So perhaps we were relying on an implementation detail (or perhaps there was a breaking change between 1.x and 2.x). There's currently a ticket tracking this (https://jira.mongodb.org/browse/CSHARP-2411), so perhaps we can defer this. I've rescheduled the ticket for triage in either case

Reply:
@DmitryLukyanov would you be able to confirm that the old test passes under .NET Core 1.0?

Reply:
I can confirm that the order is different for `NETCOREAPP2_1` and other frameworks. And I can confirm, that the test with the currently added solution passes. I will check this issue closely and will come back with details.